engine/kv_engine.cpp

```diff
@@ -1051,6 +1051,9 @@ Status KVEngine::SDeleteImpl(Skiplist* skiplist, const StringView& user_key) {
 
     if (!need_write_delete_record) {
       if (sized_space_entry.size > 0) {
+        // We must mark a allocated but unused space on PMem, otherwise data may
+        // lost in recovery
+        markEmptySpace(sized_space_entry);
```

Comment:
It is surprising behavior for programmer that if you malloc(), you cannot directly free().
Also, this adds mental overhead to track all the situations in which we must maskEmptySpace().